lib/puma/cluster/worker.rb

```diff
@@ -165,7 +166,7 @@ def spawn_worker(idx)
                                   launcher: @launcher,
                                   pipes: { check_pipe: @check_pipe,
                                            worker_write: @worker_write },
-                                  server: @server
+                                  app: @server.app
```

Comment:
We can't simply use `start_server` and pass the new server in here like we currently (on master) do in `Cluster#spawn_worker`, cause if worker 0 was given a server during initialization, with a preloaded app, it's `#app` will initialize a new app on the `start_server` call, which we don't want. Instead, we grab the preloaded app off its server and pass that through instead.

An alternative approach is to ensure `@app` is set in `Cluster#initialize` if a server is passed in. However, doing it this way means the worker always starts its own server, which imo is a better separation of responsibilities.